JASP-Desktop/widgets/qmllistview.cpp

```diff
@@ -370,7 +370,8 @@ void QMLListView::sourceChangedHandler()
 			removeDependency(sourceModel->listView());
 			disconnect(sourceModel, &ListModel::modelChanged, listModel, &ListModel::sourceTermsChanged);
 			for (SourceType& discardModel : sourceItem->discardModels)
-				disconnect(discardModel.model, &ListModel::modelChanged, listModel, &ListModel::sourceTermsChanged);
+				if (discardModel.model)
+					disconnect(discardModel.model, &ListModel::modelChanged, listModel, &ListModel::sourceTermsChanged);
```

Comment:
Here it is a bit reasonable to check the model for being null, but you could also let the connecting be done by `SourceType` itself and simply passing the listModel as a pointer. And then telling `discardModel.connectSourceTermsChanged(listModel)`. 

Hmm.